src/Interpreters/ActionsDAG.h

```diff
@@ -347,6 +347,9 @@ class ActionsDAG
         const std::unordered_map<std::string, ColumnWithTypeAndName> & node_name_to_input_node_column,
         const ContextPtr & context);
 
+    /// Return original node (input) for alias in output if exists
+    const Node * getOriginalNodeForOutputAlias(const String & output_name);
```

Comment:
It looks a bit specific to me, and can be implemented externally.
Maybe let's move it out of ActionsDAG?

Reply:
Can do it, np, but guess it'll end up here anyway because new analyzer always create aliases on original columns, and sometimes we need to understand (not only here) if an alias refers to a particular column or not.

Reply:
Let's, please, move it for outside now :D